corehq/apps/es/users.py

```diff
@@ -233,7 +233,13 @@ def is_active(active=True):
 
 
 def user_data(key, value, domain=None):
+    if isinstance(value, list):
+        return filters.OR([_user_data_filter(key, v, domain=domain) for v in value])
+    else:
+        return _user_data_filter(key, value, domain=domain)
+
 
+def _user_data_filter(key, value, domain=None):
```

Comment:
This uses `queries.match`, which seemed like an odd choice, assuming we're generally looking for exact matches, not similar terms.  Looks like I made that decision deliberately back in
https://github.com/dimagi/commcare-hq/pull/28667
Though I'm not sure I'm swayed by my reasoning at the time.  What do you think?  Out of scope for this PR sure, but I'm wondering whether switching from a `term` to a `match` results in functional changes for the login-as queries here.

Reply:
Yeah, I had similar thoughts, but kept it as it is to avoid any functional regressions. It could be a separate task for latter.